src/components/store/ingredients.ts

```diff
@@ -0,0 +1,57 @@
+import { StateCreator } from "zustand";
+import { NewIngredient } from "../types.ts";
+
+interface MyIngredientsActions {
+  addIngredient: (ingredient: NewIngredient) => void;
+  editIngredients: (ingredients: NewIngredient[]) => void;
+  editIngredientsBoughtOften: (
+    ingredients: {
+      id: NewIngredient["id"];
+      amount: number;
+      frequency: number;
+    }[]
+  ) => void;
+}
+
+interface MyIngredientsState {
+  ingredients: NewIngredient[];
+
+  ingredientsBoughtOften: {
+    id: NewIngredient["id"];
+    amount: number;
+    frequency: number;
+  }[];
+}
+
+const initialIngredientsState: MyIngredientsState = {
+  ingredients: [],
+  ingredientsBoughtOften: [],
+};
+
+export const createIngredientsSlice: StateCreator<
+  MyIngredientsState & MyIngredientsActions,
+  [],
+  [],
+  MyIngredientsState & MyIngredientsActions
```

Comment:
Jen pro zajimavost (co se TypeScriptu tyce)

ten prvni radek rika, co pouzivas
ten posledni, co poskytujes 

je to trochu crypticke, tak jen kdyz jsem v tom, at ti to dava smysl...

Teoreticky neni problem udelat slice, co "kombinuje dalsi dva slices". V tom set mas pristup ke vsemu, jen ti to typescript nerekne, pokud mu neporadis. Zatim to ale asi nepotrebujem.